stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -467,25 +466,21 @@ public class ChatClient internal constructor(
         offset: Int,
         limit: Int,
         type: String
-    ): Call<List<AttachmentWithDate>> {
+    ): Call<List<Attachment>> =
+        getMessagesWithAttachments(channelType, channelId, offset, limit, type).map { messages ->
+            messages.flatMap { message -> message.attachments.filter { it.type == type } }
+        }
+
+    public fun getMessagesWithAttachments(
```

Comment:
It might be worth it to add KDOC explaining that this will return messages which contain at least one desired type attachment and not necessarily all of them will have a specified type.